code/modules/projectiles/projectile/energy/stun.dm

```diff
@@ -2,28 +2,343 @@
 	name = "electrode"
 	icon_state = "spark"
 	color = COLOR_YELLOW
-	paralyze = 10 SECONDS
-	stutter = 10 SECONDS
-	jitter = 40 SECONDS
 	hitsound = 'sound/items/weapons/taserhit.ogg'
-	range = 7
+	range = 5
 	tracer_type = /obj/effect/projectile/tracer/stun
 	muzzle_type = /obj/effect/projectile/muzzle/stun
 	impact_type = /obj/effect/projectile/impact/stun
+	/// How much stamina damage will the tase deal in 1 second
+	var/tase_stamina = 60
 
 /obj/projectile/energy/electrode/on_hit(atom/target, blocked = 0, pierce_hit)
 	. = ..()
-	if(!ismob(target) || blocked >= 100) //Fully blocked by mob or collided with dense object - burst into sparks!
-		do_sparks(1, TRUE, src)
-	else if(iscarbon(target))
-		var/mob/living/carbon/C = target
-		C.add_mood_event("tased", /datum/mood_event/tased)
-		SEND_SIGNAL(C, COMSIG_LIVING_MINOR_SHOCK)
-		if(HAS_TRAIT(C, TRAIT_HULK))
-			C.say(pick(";RAAAAAAAARGH!", ";HNNNNNNNNNGGGGGGH!", ";GWAAAAAAAARRRHHH!", "NNNNNNNNGGGGGGGGHH!", ";AAAAAAARRRGH!" ), forced = "hulk")
-		else if(!C.check_stun_immunity(CANKNOCKDOWN))
-			addtimer(CALLBACK(C, TYPE_PROC_REF(/mob/living/carbon, do_jitter_animation), 20), 0.5 SECONDS)
+	if(pierce_hit)
+		return .
+	do_sparks(1, TRUE, src)
+	if(. == BULLET_ACT_BLOCK || !isliving(target) || blocked >= 100)
+		visible_message(span_warning("[src]\s fail to shock [target], falling to [get_turf(target)]."))
+		return .
+
+	var/mob/living/tased = target
+	if(tased.apply_status_effect(/datum/status_effect/tased, fired_from, firer, tase_stamina, null, "\the [src]\s"))
+		return .
+	visible_message(span_warning("[src]\s fail to shock [target], falling to [get_turf(target)]."))
```

Comment:
I already noticed this in the video, but this probably parses very weirdly. Notably, it said '...falling to the reinforced wall.' when an electrode hit a wall.

Honestly, unless this has any mechanical relevancy, you may as well not have that last part.

Reply:
There's ``last_tick_turf`` which could be used for this, but I agree with necro - i don't think that this is particularly necessary.